lib/extreme.ex

```diff
@@ -83,7 +98,7 @@ defmodule Extreme do
   @doc """
   TODO
   """
-  @callback execute(message :: term, correlation_id :: UUID.t()) :: term
+  @callback execute(message :: term(), correlation_id :: binary(), timeout :: integer()) :: term()
```

Comment:
from what I've seen elsewhere, specs and arities typically align with the widest arity a function supports (e.g. [`GenServer.call/3`](https://hexdocs.pm/elixir/GenServer.html#call/3) describes 3 parameters, even though one is defaulted). (although I don't think this pattern is documented or formalised anywhere 😬 )

Reply:
good point!